integration-tests/itest/deploy_test.go

```diff
@@ -0,0 +1,306 @@
+			Expect(err).NotTo(HaveOccurred())
+			appNames := listAppNames(list)
+			Expect(appNames).To(ContainElement(swissKnifeApp.Name))
+		})
+
+		It("should get to a 'running' status, eventually", func(done Done) {
+			// TODO: check status flag once it's fixed
+			status, err := checkStatusIs(swissKnifeApp, "Running")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(status).To(Equal("Running"))
+			close(done)
+		}, LongTimeout)
+	})
+
+	Context("A deployed test application that uses akka, spark, and flink", func() {
+		It("should contain a spark process", func() {
+			status, err := cli.Status(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			streamlets := cli.GetStreamlets(&status)
+			Expect(streamlets).To(ContainElement("spark-process"))
+		})
+
+		It("should contain a flink process", func() {
+			status, err := cli.Status(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			streamlets := cli.GetStreamlets(&status)
+			Expect(streamlets).To(ContainElement("flink-process"))
+		})
+
+		It("should contain an akka process", func() {
+			status, err := cli.Status(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			streamlets := cli.GetStreamlets(&status)
+			Expect(streamlets).To(ContainElement("akka-process"))
+		})
+
+		It("should produce a counter in the raw output log", func(done Done) {
+			pod, err := cli.GetSinglePodForStreamlet(swissKnifeApp, "raw-egress")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = checkLastLogsContains(pod, swissKnifeApp.Name, "count:")
+			Expect(err).NotTo(HaveOccurred())
+			close(done)
+		}, LongTimeout)
+
+		It("should produce a counter in the akka output log", func(done Done) {
+			pod, err := cli.GetSinglePodForStreamlet(swissKnifeApp, "akka-egress")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = checkLastLogsContains(pod, swissKnifeApp.Name, "count:")
+			Expect(err).NotTo(HaveOccurred())
+			close(done)
+		}, LongTimeout)
+
+		It("should produce a counter in the spark output log", func(done Done) {
+			pod, err := cli.GetSinglePodForStreamlet(swissKnifeApp, "spark-egress")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = checkLastLogsContains(pod, swissKnifeApp.Name, "count:")
+			Expect(err).NotTo(HaveOccurred())
+			close(done)
+		}, LongTimeout)
+
+		It("should produce a counter in the flink output log", func(done Done) {
+			pod, err := cli.GetSinglePodForStreamlet(swissKnifeApp, "flink-egress")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = checkLastLogsContains(pod, swissKnifeApp.Name, "count:")
+			Expect(err).NotTo(HaveOccurred())
+			close(done)
+		}, LongTimeout)
+
+	})
+
+	Context("A deployed application can be undeployed", func() {
+		It("should undeploy the test app", func(done Done) {
+			err := cli.Undeploy(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			err = expectFindApp(swissKnifeApp, false)
+			Expect(err).NotTo(HaveOccurred())
+			close(done)
+		}, ShortTimeout)
+	})
+})
+
+func expectFindApp(app cli.App, expected bool) error {
```

Comment:
I suggest we move all of these utils to a Golang package since they all look like general test utils instead of being specific for this test.

Reply:
I'd argue that `ListApps()` belongs to the package and this method is a helper method to implement the specific test logic, in this case, assert that the app is in the list. (in scala: `ListApps().exist(e => e.name == app.name)`)
I think that moving it into the pkg would pollute it as each test will contribute to such noise. But I'm a Go noob.
What do other team members think? @olofwalker @RayRoestenburg @skonto

Reply:
I would prefer to have tests in one package and support for tests in one.

Reply:
@olofwalker currently it's structured like this:
```
test suite
|-- cli (supporting lib. Maybe add kctl next to it)

```
Do you suggest to reorganize it as:
```
main (noop)
 |- test suite
 |- cli
 |- kctl (tbc)
```
?
I particularly struggle with supporting functions, such as map-like transformations (e.g. `GetStreamletNames(StreamletPods)`. It seems silly to put them in a separate package.
Or maybe it's the way to _GO_ ?

Reply:
But `expectFindApp` and `checkLastLogsContains` are clearly general purpose utils that will be reused in other tests, right? There is nothing specific to this particular tests in the logic for those functions or others.

So unless you think that this one spec will be the only spec we will need to perform those checks, I think they should be more separate.

Reply:
I'm fine with waiting a little bit until we write  second test but what I want to prevent is dozens of minor variations of these util functions or one tests needing to import another test to gain access to its utils.

Reply:
I reorganized the test and helper classes to reduce noise as much as possible.